WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductVariationSelectorViewModel.swift

```diff
@@ -159,10 +168,19 @@ final class ProductVariationSelectorViewModel: ObservableObject {
             return
         }
 
-        if let onVariationSelected = onVariationSelected {
-            onVariationSelected(selectedVariation, parentProduct)
+        if shouldToggleItemOnSelection {
+            // Product variation multi-selection (new)
+            if let onVariationSelected = onVariationSelected {
+                toggleSelection(productVariationID: variationID)
+                onVariationSelected(selectedVariation, parentProduct)
+            }
         } else {
-            toggleSelection(productVariationID: variationID)
+            // Product variation single-selection (legacy)
+            if let onVariationSelected = onVariationSelected {
+                onVariationSelected(selectedVariation, parentProduct)
+            } else {
+                toggleSelection(productVariationID: variationID)
+            }
```

Comment:
💡  Same comment as above for readability.

Reply:
Changed on f06999092d5225f5a9a72a2f2c2cf1211ff286fd